src/lib/swapper/api.ts

```diff
@@ -55,10 +60,11 @@ type ChainSpecificQuoteFeeData<T extends ChainId> = ChainSpecific<
   }
 >
 
+export type ProtocolFee = { requiresBalance: boolean } & AmountDisplayMeta
```

Comment:
ProtocolFee also includes `asset` as a property in case an unsupported asset appears in the quote. This allows us to handle/render it downstream without resorting to "unknown asset" type of logic